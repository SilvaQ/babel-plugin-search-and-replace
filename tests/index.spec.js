const babel = require("babel-core");
const searchAndReplaceplugin = require("../src/index");

var example = `
var foo = 1;
if (foo) console.log(foo);
const hello = "hello";

const myFunc = function(param = "hello"){
    return 1;
}
`;

it("should replace nothing", () => {
  const { code } = babel.transform(example, {
    plugins: [searchAndReplaceplugin]
  });
  expect(code).toMatchSnapshot();
});

it("should replace foo by bar", () => {
  const { code } = babel.transform(example, {
    plugins: [
      [
        searchAndReplaceplugin,
        {
          option1: true,
          option2: false
        }
      ]
    ]
  });
  expect(code).toMatchSnapshot();
});
