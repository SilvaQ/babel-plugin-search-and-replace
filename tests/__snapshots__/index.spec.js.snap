// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should replace foo by bar 1`] = `
"
var bar = 1;
if (bar) console.log(bar);
const hello = \\"hello\\";

const myFunc = function (param = \\"hello\\") {
    return 1;
};"
`;

exports[`should replace foo by bar and hello by hey 1`] = `
"
var bar = 1;
if (bar) console.log(bar);
const hey = \\"hello\\";

const myFunc = function (param = \\"hello\\") {
    return 1;
};"
`;

exports[`should replace nothing 1`] = `
"
var foo = 1;
if (foo) console.log(foo);
const hello = \\"hello\\";

const myFunc = function (param = \\"hello\\") {
    return 1;
};"
`;
